src/GHC.hs

```diff
@@ -107,9 +107,10 @@ stage2Packages = return [haddock]
 testsuitePackages :: Action [Package]
 testsuitePackages = return [ checkApiAnnotations
```

Comment:
We have this list of packages needed for the testsuite, but then do a lot of ad hoc `need`s in the testsuite instead of fully relying on this list. Can we do better?

Reply:
Actually, default libraries are built vanilla way, but the libraries listed here are needed in profiling and dynamic ways also. Can we create a separate list of packages needed only  for testsuite and have to built in profiling and dynamic ways.

Reply:
So, maybe this list should list ways too?

I think someone needs to think about the overall architecture of the testsuite. So far there are a bunch of PRs that seem to introduce various ad hoc solutions, but I do not see a clear design of how everything should be tied together. Can you write up such a design document?

Reply:
Yes, I will be creating an Issue in this regards soon enough.

Reply:
Yes we need to be a _lot_ more careful about those uses of `vanillaContext` and default ways for packages etc. Most of the time we just want to follow what `libraryWays` and `rtsWays` dictate, I think. I'm expecting this single line of work to fix a _lot_ of test failures.